Vagrantfile

```diff
@@ -46,6 +46,15 @@ else
     Vagrant.require_version ">= 1.6.2"
 end
 
+synced_folder_type = custom_config.get('synced_folder_type')
+
+unless (synced_folder_type == "nfs" || synced_folder_type == "virtualbox")
+    class SyncedFolderTypeError < Vagrant::Errors::VagrantError
+        self.error_message "[synced_folder_type] Only 'nfs' and 'virtualbox' are supported values"
+    end
+    raise SyncedFolderTypeError
+end
+
```

Comment:
The suggested way by Vagrant to raise expected errors: https://github.com/mitchellh/vagrant/blob/master/lib/vagrant/errors.rb
I could not subclass Vagrant::Errors::ConfigInvalid as this causes an i18n error on my machine (vagrant up does depend on the host) and I don't have time to investigate.